arangod/Replication2/StateMachines/Document/DocumentStateStrategy.cpp

```diff
@@ -45,10 +45,10 @@
 
 using namespace arangodb::replication2::replicated_state::document;
 
-DocumentStateAgencyHandler::DocumentStateAgencyHandler(GlobalLogIdentifier gid,
-                                                       ArangodServer& server,
-                                                       AgencyCache& agencyCache)
-    : _gid(std::move(gid)), _server(server), _agencyCache(agencyCache) {}
+DocumentStateAgencyHandler::DocumentStateAgencyHandler(
+    GlobalLogIdentifier gid, ArangodServer& server,
+    ClusterFeature& clusterFeature)
+    : _gid(std::move(gid)), _server(server), _clusterFeature(clusterFeature) {}
```

Comment:
I did some changes which required me to modify the constructor, because I wanted to write to the agency. I then reverted the changes, but left the constructor as it is now because in the current stage of development it's good to have the entire clusterFeature in there, for debugging purposes. I will add a TODO, so we optimize this dependency eventually.